### DISCUSSION TITLE
is it possible to send a HTML file as response to a request?

### DISCUSSION THREAD

**Discussion by unnaturalistic** [vercel/next.js#47129] [Help]
### Summary

i have a pages folder and i have a file named ` about.js` , the code for about.js is as following

```
export default function About() {
    return (
    <>
    <div>About</div>
    <p>haha it works!</p>
    </>
    );
}
```

is it possible to return a html file instead of making JSX responses?
IE: 
`return ('about.html')`

**Comment by maxymapp** [comment]
You already have `pages/about.js` that is rendered as an HTML file.
The below code wouldn't make any sense even if it worked.
```
export default function About() {
   return ('about.html')
}
```
Why would you need to return the HTML page from within itself?
I guess you meant to ask your question a bit differently: _"How do I serve an HTML file as a response in the code (without Next.js file system-based routing)?"_
**I think you might want to escape the default routing, and tell the framework when your HTML file should be served.
Here is a good answer: https://articles.wesionary.team/render-html-file-in-pages-of-next-js-59281c46c05**
I hope it works for you!

**Reply by unnaturalistic** [reply]
yes i meant that in my question
i will look at the solution you provided
